### ISSUE TITLE
N64 Games Can Hang Forever with "Sync to Audio" & a low buffer

### ISSUE THREAD

**Issue by NarryG** [TASVideos/BizHawk#1240]
When utilizing "Sync to Audio", Star Wars EP1 racer will crash Bizhawk if the buffer is too low. Using Sync to Clock makes the game stutter like mad so you pretty much need to use Sync to Audio. I had previously lowered my buffer to the minimum of 60ms which has worked fine in many other situations, but in this game even the default of 100 is being problematic on DirectSound

**Comment by NarryG** [comment]
To be more clear, it's an infinite hang due to this code.
```					while (samplesProvided > samplesNeeded)
					{
						Thread.Sleep((samplesProvided - samplesNeeded) / (SampleRate / 1000)); // Let the audio clock control sleep time
						samplesNeeded = _outputDevice.CalculateSamplesNeeded();
					}```
The specific issue is that samplesProvided is always above samplesNeeded.

**Comment by NarryG** [comment]
This issue appears to apply to other games as well. DK64 being another example.

**Comment by jdpurcell** [comment]
I am closing this because it appears to have been fixed by commit 2b53f9b9.
